src/reactor/fns.js

```diff
@@ -0,0 +1,444 @@
+import Immutable from 'immutable'
+import logging from '../logging'
+import { isImmutableValue } from '../immutable-helpers'
+import { toImmutable } from '../immutable-helpers'
+import { fromKeyPath, getStoreDeps, getComputeFn, getDeps, isGetter } from '../getter'
+import { isEqual, isKeyPath } from '../key-path'
+import { each } from '../utils'
+
+/**
+ * Immutable Types
+ */
+var EvaluateResult = Immutable.Record({ result: null, reactorState: null})
```

Comment:
Awesome work on the rewrite @jordangarcia!

This file still contains some `var` declarations. Not sure if you care but I saw that you converted most to `let` and `const` so wanted to let you know.